Code/Framework/AzFramework/AzFramework/Asset/AssetProcessorMessages.h

```diff
@@ -1254,6 +1254,44 @@ namespace AzFramework
             FolderList m_folderList;
         };
 
+        class AssetChangeReportRequest
+            : public BaseAssetProcessorMessage
+        {
+        public:
+            AZ_CLASS_ALLOCATOR(AssetChangeReportRequest, AZ::OSAllocator, 0);
+            AZ_RTTI(AssetChangeReportRequest, "{EF108E73-08F7-4CBC-A808-61A2EC544A6E}", BaseAssetProcessorMessage);
+            static void Reflect(AZ::ReflectContext* context);
+            static constexpr unsigned int MessageType = AZ_CRC_CE("AssetSystem::AssetChangeReport");
+
+            enum ChangeType
+            {
+                CheckMove,
+                Move
+            };
+
+            AssetChangeReportRequest() = default;
+            AssetChangeReportRequest(const AZ::OSString& fromPath, const AZ::OSString& toPath, ChangeType changeType = ChangeType::Move);
+            unsigned int GetMessageType() const override;
+
+            AZ::OSString m_fromPath;
+            AZ::OSString m_toPath;
+            ChangeType m_type;
+        };
+
+        class AssetChangeReportResponse
+            : public BaseAssetProcessorMessage
+        {
+        public:
+            AZ_CLASS_ALLOCATOR(AssetChangeReportResponse, AZ::OSAllocator, 0);
+            AZ_RTTI(AssetChangeReportResponse, "{C18891A7-794D-4270-93AE-7D0C2ECABB5C}", BaseAssetProcessorMessage);
+            static void Reflect(AZ::ReflectContext* context);
+
+            AssetChangeReportResponse() = default;
```

Comment:
Nitpick: a comment stating that this is required by the SerializeContext and isn't intended for use otherwise could be useful, if that's the case.

Reply:
Done.